app/services/Agents.scala

```diff
@@ -104,12 +104,21 @@ class Agents @Inject() (amiableConfigProvider: AmiableConfigProvider, lifecycle:
   }
 
   def refreshOldInstanceCountInfo(instancesWithAmis: List[(Instance, Option[AMI])]): Unit = {
-    val now = DateTime.now
-    val oldInstanceCountsByAccount = PrismLogic.oldInstances(instancesWithAmis)
-      .groupBy(_.meta.origin.accountName.getOrElse("unknown-account"))
-      .map{ case (accountName, list) => OldInstanceAccountHistory(now, accountName, list.length)}
-      .toList
-    oldInstanceCountByAccountAgent.send(oldInstanceCountsByAccount)
+    for {
+      accounts <- Prism.getAccounts
+    } yield {
+      val now = DateTime.now
+      val amisForAccount = instancesWithAmis.groupBy(_._1.meta.origin.accountName.getOrElse("unknown-account"))
+
+      val oldInstanceCountsByAccount = accounts.map( account => {
```

Comment:
Another way of writing this - though not massively tidier
```
accounts.map { account =>
   val oldInstanceCount = amisForAccount.get(accoount.accountName).map(PrismLogic.oldInstances).getOrElse(0)
OldInstanceAccountHistory(now, account.accountName, oldInstanceCount)
}
```

Reply:
actually maybe even cleaner you could call PrismLogic on instancesWithAmis before the group by as was done before, not certain if that  would work though!

Reply:
It does work! Good shout